02_bubble/sirigithub/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubbleSort(s []int) []int {
+	n := len(s)
+	arrayToSort := make([]int, n)
+	copy(arrayToSort, s)
+
+	for i := 0; i < n; i++ {
+		for j := i; j > 0 && arrayToSort[j-1] > arrayToSort[j]; j-- {
+			arrayToSort[j-1], arrayToSort[j] = arrayToSort[j], arrayToSort[j-1]
+		}
+	}
+	return arrayToSort
+}
+
+func insertionSort(s []int) []int {
+	n := len(s)
+	arrayToSort := make([]int, n)
+	copy(arrayToSort, s)
+
+	for i := 1; i < n; i++ {
+		for j := i; j > 0 && arrayToSort[j-1] > arrayToSort[j]; j-- {
+			arrayToSort[j-1], arrayToSort[j] = arrayToSort[j], arrayToSort[j-1]
+		}
+	}
```

Comment:
This is identical to the bubble sort earlier. It looks slightly different with `i` starting at `1`, but as per my previous comment, that first iteration does nothing so it may as well start at `1`.

The wikipedia pages for bubble sort and insertion sort have some pseudo code you can model your code on.

Reply:
Done. Referenced implementation as per wiki.